ShoppingList/ShoppingList/View Controllers/AddTaskViewController.swift

```diff
@@ -0,0 +1,236 @@
+//
+//  AddTaskViewController.swift
+//  ShoppingList
+//
+//  Created by Jerrick Warren on 5/20/19.
+//  Copyright © 2019 Lambda School Labs. All rights reserved.
+//
+
+import UIKit
+
+class AddTaskViewController: UIViewController, UITableViewDataSource, UITableViewDelegate, UITextFieldDelegate {
+   
+
+    // Load sample values
+    
+    
+    
+    var sampleCategories = ["Kitchen", "Bathroom", "Dining Room", "Living Room" ,"Master Bedroom"]
+    var sampleTask = ["Clean up", "Wipe Down", "Wash dishes", "Vaccum", ]
+    
+    // Properties
+    var categories: [Category]?
+    var userController: UserController?
+    var currentUser: User?
+    var householdController: HouseholdController?
+    var taskController = TaskController()
+    var categoryController = CategoryController()
+    
+    var task : Task?
+    var catID : UUID?
+    var category : Category?
+    var household : Household?
+    
+    //∫???
+    // let tabBarController = TabViewViewController.self
+    
+    let mainCategoriesViewController = MainCategoriesViewController()
+    
+    // category reuse Identifier
+    
+    let reuseIdentifier = "searchCategoryCell"
+    
+    var filteredResults : [Category] = [] {
+        didSet {
+            DispatchQueue.main.async {
+                self.categoryTableView.reloadData()
+            }
+        }
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        
+        fetchCategories()
+    }
+    
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        categoryTableView.delegate = self
+        categoryTableView.dataSource = self
+        addCategoryTextField.delegate = self
+        
+        categoryTableView.rowHeight = UITableView.automaticDimension
+        
+        if let tabBar = self.tabBarController as? TabViewViewController {
+        
+            self.taskController = tabBar.taskController
+            self.currentUser = tabBar.currentUser
+            self.householdController = tabBar.householdController
+            self.userController = tabBar.userController
+            
```

Comment:
We will set categoryController from the tabBar on this line. @jWarrenDev

Reply:
ok Cool